### ISSUE TITLE
Blobs not shown until searched

### ISSUE THREAD

**Issue by jussivesa** [microsoft/AzureStorageExplorer#1264]
Repro:
- Save new blob files
- Refresh
- **Sort by date**
- Not all blobs show
- Search with prefix
- Now the blobs are actually show that initially were missing :(

See Img 1 vs Img 2. Image 2 has the blobs that are now shown in the full list.

Img 1
![bug-1](https://user-images.githubusercontent.com/17849758/55316409-fe401f00-5476-11e9-856c-622db82859b0.PNG)

Img 2
![bug-2](https://user-images.githubusercontent.com/17849758/55316427-08621d80-5477-11e9-8ce4-307435b813d8.PNG)

**Comment by MRayermannMSFT** [comment]
Hi @jussivesa , unfortunately this is by design. When you sort by date we only sort the blobs we have already cached. So in your case, the blobs you are looking for have probably not yet been retrieved from the server and added to our cache. But when you do the prefix search, that is an actually API level filter, so the cache is updated to include the blobs you are looking for. We've been wanting to make the column sorts an API filter for a while, but we're waiting on the platform team to implement that feature for us.

**Comment by jussivesa** [comment]
Hi @MRayermannMSFT. Hmm, that's quite bad design IMO. It took me good couple of hours of debugging while trying to figure out why my logs said that new blobs were saved until I did take a look in the Azure portal view of the blobs. 

What does the refresh button do, if not "bust" the cache? I believe that the root cause of this is the cache on Azures end, but it felt like a bug when all except one blob with prefixed name `Ring-` were showing in the "all items list" after new files were saved and the refresh button was pressed.

Glad to hear that you're implementing a fix for this issue. It would be nice if the interface would state something like "Azure cache last updated xx.xx.xxxx xx:xx:xx" when not using the API level filter. :)

**Comment by MRayermannMSFT** [comment]
I would say that at this point we can close this issue. As of version 1.32.0, Storage Explorer now offers a variety of settings which can make this situation not happen. The are:

**Global Sort**

When global sort is enabled, sorting by column will load all data for the current folder or view and then sort. While data is being loaded, progress can be monitored in the notification bar at the top of the explorer. If the load is taking too long, you can cancel it by clicking the cancel button in the notification bar. Global sort can be enabled by going to Settings → Data Explorers → Global Sort.

**Load All**

When load all is enabled, loading more (e.g. clicking "Load More") will load all available data for the current view. Just like for global sort, progress can be monitored and canceled via the notification bar at the top of the explorer. Load all can be enabled by going to Settings → Data Explorers → Load All.

 **Load Batch Size**

The load batch size setting controls how many items are loaded per batch whenever a data explorer fetches data from the server. This includes initial loading of a view, refreshing, and when clicking load more (if load all is not enabled). This setting can be found at Settings → Data Explorers → Load Batch Size.

If these features do not resolve this issue, please open a new issue or leave feedback on this [issue](https://aka.ms/storageexplorer/globalsortfeedback).

Thanks!
